src/initialization/MOM_grid_initialize.F90

```diff
@@ -255,16 +283,24 @@ subroutine set_grid_metrics_from_mosaic(G, param_file, US)
   deallocate(exni)
   deallocate(exnj)
 
-  ! Read X from the supergrid
-  tmpZ(:,:) = 999.
-  call MOM_read_data(filename, 'x', tmpZ, SGdom, position=CORNER)
+  ! open the file for domain-decomposed read
+  file_open_success = open_file(fileObjRead, filename, "read", SGdom%mpp_domain, is_restart=.false.)
+  ! tmpZ is defined on the data domain
+  tmpZ(:,:) = 999.0
+  ! register the global axes
+  call register_axis(fileObjRead, 'nxp','x', domain_position=EAST_FACE)
```

Comment:
nxp should be something that can be inquired for. The names of dimensions should not be hard coded for a read.

Reply:
This potential solution queries the file for the global dimension names, loops through each one, and registers the dimensions that match the sizes of temporary buffers that will hold each array. 

```
ndims = get_num_dimensions(fileObjRead)
  allocate(dimNames(ndims))
  call get_dimension_names(fileObjRead, dimNames)
  ! register the global axes
  do i=1,ndims

    call get_dimension_size(fileObjRead, dimNames(i), globalDimSize)

    if (globalDimSize .eq. size(tmpz,1)) then
      call register_axis(fileObjRead, trim(dimNames(i)),'x', domain_position=CENTER)
    elseif (globalDimSize .eq. size(tmpz,2)) then
      call register_axis(fileObjRead, trim(dimNames(i)),'y', domain_position=CENTER)
    elseif (globalDimSize .eq. size(tmpV,1)) then
      call register_axis(fileObjRead, trim(dimNames(i)),'x', domain_position=EAST_FACE)
    elseif (globalDimSize .eq. size(tmpU,2)) then
      call register_axis(fileObjRead, trim(dimNames(i)),'y', domain_position=NORTH_FACE)
    endif
  enddo

  deallocate(dimNames)
```